src/main/java/team/reborn/energy/Energy.java

```diff
@@ -1,19 +1,25 @@
 package team.reborn.energy;
 
-import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.stream.Stream;
 
 public final class Energy {
 
-	private static final HashMap<Predicate<Object>, Function<Object, EnergyStorage>> holderRegistry = new HashMap<>();
+	@SuppressWarnings("rawtypes")
+	private static final List<EnergyHolderProvider> PROVIDERS = new ArrayList<>();
+	@SuppressWarnings("rawtypes")
+	private static final Predicate TRUE = o -> true;
 
 	static {
-		registerHolder(object -> object instanceof EnergyStorage, object -> (EnergyStorage) object);
+		registerHolder(EnergyStorage.class, -5, object -> object);
```

Comment:
why -5 as default and not 0?

Reply:
I wanted to have holder providers be higher than the defaulted ones by default